### PR TITLE
[SH4RE-98] Convert string availability to dates

### PR DESCRIPTION
**Pull request by WasifButt**
- Handle everyday, weekday, weekend and convert them to an array of dates 
- Change 1 year calendar to a quarter (3 months)

### CODE REVIEW

File: SH4RE/SH4RE/CreateListing/CreateListingView.swift [hunk 1]
```diff
@@ -178,7 +176,7 @@ struct CreateListingView: View {
                 .padding(.bottom)
             
             // availability
-            DropdownMenu(label: "Availability", options: availabilityList, selection: $availabilitySelection)
+            DropdownMenu(label: "Availability for the next 3 months", options: availabilityList, selection: $availabilitySelection)
```
Comment (americobarros) [inline SH4RE/SH4RE/CreateListing/CreateListingView.swift]: I think having "for the next 3 months" is a liiiiittle strange but I also get why you did it, and don't see too much of a way around it. Ideally, we would have had a system that made it available every day forever then only allow bookings to be made in the next 3 months for example, but this is okay for demo purposes I guess.
Also, plz make it capitalized. I think Cost Per Day should have also been capitalized. So "Availability For The Next 3 Months" or "Availability for the Next 3 Months" or some variation of that

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift
Comment (americobarros) [inline SH4RE/SH4RE/ViewListing/ViewListingView.swift:1]: Screenshot 2023-03-24 at 4 15 49 PM
getting an errror on building preview for this file now

File: SH4RE/SH4RE/CreateListing/CreateListingView.swift [hunk 2]
```diff
@@ -226,6 +224,20 @@ struct CreateListingView: View {
             }
             else {
                 calAvail.append(availabilitySelection)
```
Comment (americobarros) [inline SH4RE/SH4RE/CreateListing/CreateListingView.swift]: maybe this should be moved to the end of the if, else if, .. , else {} just for clarity sake

File: SH4RE/SH4RE/CreateListing/CreateListingView.swift [hunk 3]
```diff
@@ -226,6 +224,20 @@ struct CreateListingView: View {
...
+                let calendar = Calendar.current
+                let components = calendar.dateComponents([.year, .month, .day], from: Date())
+                let startOfMonth = calendar.date(from:components)!
+                let numberOfDays = calendar.range(of: .day, in: .quarter, for: startOfMonth)!.upperBound
+                let allDays = Array(0..<numberOfDays).map{ calendar.date(byAdding:.day, value: $0, to: startOfMonth)!}
+                if (availabilitySelection == "Everyday") {
+                    calAvail = []
+                }
+                else if (availabilitySelection == "Weekdays") {
+                    calAvail = allDays.filter{ calendar.isDateInWeekend($0) }
```
Comment (americobarros) [inline SH4RE/SH4RE/CreateListing/CreateListingView.swift:235]: why are we filtering the dates that ARE in the weekend, if the availability is set to weekdays?

File: SH4RE/SH4RE/CreateListing/CreateListingView.swift [hunk 4]
```diff
@@ -226,6 +224,20 @@ struct CreateListingView: View {
...
+                }
+                else if (availabilitySelection == "Weekends") {
+                    calAvail = allDays.filter{ !calendar.isDateInWeekend($0) }
```
Comment (americobarros) [inline SH4RE/SH4RE/CreateListing/CreateListingView.swift:238]: same as above, why grab dates that are NOT in the weekend, if avail is weekends?

**Review by americobarros — CHANGES_REQUESTED** [review]

File: SH4RE/SH4RE/CreateListing/CreateListingView.swift (hunk 3 above)
Comment (WasifButt) [reply SH4RE/SH4RE/CreateListing/CreateListingView.swift:235]: it's just cause of the way it's handled on view listing side. The dates that it retrives is the dates to **block**, so if we want weekdays we want to store weekends in database cause those will be blocked when creating availbility during view listing

File: SH4RE/SH4RE/CreateListing/CreateListingView.swift (hunk 4 above)
Comment (WasifButt) [reply SH4RE/SH4RE/CreateListing/CreateListingView.swift:238]: see above

File: SH4RE/SH4RE/CreateListing/CreateListingView.swift (hunk 1 above)
Comment (WasifButt) [reply SH4RE/SH4RE/CreateListing/CreateListingView.swift]: will change to "Availability for the Next 3 Months"

File: SH4RE/SH4RE/CreateListing/CreateListingView.swift (hunk 2 above)
Comment (WasifButt) [reply SH4RE/SH4RE/CreateListing/CreateListingView.swift]: that's not supposed to be there, thanks for catching

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift
Comment (WasifButt) [reply SH4RE/SH4RE/ViewListing/ViewListingView.swift:1]: that's not cause of this PR, this same behaviour is on the main branch as well. I don't think it's time efficient for me to try to fix this within this PR just for the preview. A ticket could be created for this

**Pull request merged**